MagazineLayout/Public/MagazineLayout.swift

```diff
@@ -612,17 +612,14 @@ public final class MagazineLayout: UICollectionViewLayout {
     let invalidationContext = super.invalidationContext(
       forBoundsChange: newBounds) as! MagazineLayoutInvalidationContext
 
-    invalidationContext.contentSizeAdjustment = CGSize(
-      width: newBounds.width - currentCollectionView.bounds.width,
-      height: newBounds.height - currentCollectionView.bounds.height)
     invalidationContext.invalidateLayoutMetrics = false
 
     // If our layout direction is bottom to top we want to adjust scroll position relative to the
     // bottom
     if case .bottomToTop = verticalLayoutDirection {
       invalidationContext.contentOffsetAdjustment = CGPoint(
         x: 0.0,
-        y: currentCollectionView.bounds.height - newBounds.height + contentInset.bottom)
+        y: max(currentCollectionView.bounds.height - newBounds.height + contentInset.bottom, 0))
```

Comment:
we have to limit this to positive values, otherwise when the bounds _grows_, we'll inset by a negative value (when in reality, leaving out offset alone is enough).